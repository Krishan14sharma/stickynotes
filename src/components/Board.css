.hide {
  display: none !important;
}

.notes-board {
  margin: .5em;
}
.add-note-btn {
  position: absolute;
  top: 5px;
  right: 5px;
  color: whitesmoke;
  background: #2196f3;
  border-radius: 20px;
  font-size: 22px;
  cursor: pointer;
  border-color: #42a5f5;
}

.add-note-btn:hover {
  background: #42a5f5;
}

.settings-btn {
  background: #90a4ae;
  color: whitesmoke;
  border-radius: 3px;
  font-size: 10px;
  cursor: pointer;
  position: absolute;
  top: 40px;
  right: 5px;
  font-size: 15px;
  border-radius: 20px;
  border-color: #bdbdbd;
}
.settings-btn:hover {
  background: #90a4aed4;
}

.board-settings {
  position: absolute;
  top: 40px;
  right: 40px;
  z-index: 9999999;
}

/* CSS Transition */
.note-enter {
  opacity: 0.01;
}
.note-enter.note-enter-active {
  opacity: 1;
  transition: opacity 650ms ease-in;
}

.note-leave {
  opacity: 1;
}
.note-leave.note-leave-active {
  opacity: 0.01;
  transition: opacity 750ms ease-out;
}

.note-appear {
  opacity: 0.01;
}
.note-appear.note-appear-active {
  opacity: 1;
  transition: opacity .5s ease-in;
}
