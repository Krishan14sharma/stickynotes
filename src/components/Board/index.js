import React from 'react';
import ReactCSSTransitionGroup from 'react-addons-css-transition-group';
import Note from 'Note';
import { EmojiIcon } from 'reuseables';
import SettingsPanel from 'SettingsPanel';
import * as storage from 'storage_api';
import {
	createNote,
} from 'helpers';
import {
	ThemeContext,
	NIGHT_BG_COLOR,
	DEFAULT_STORAGE_STATE,
} from 'CONSTANTS';

import './styles.css';

export default class NotesBoard extends React.Component {
	state = {
		...DEFAULT_STORAGE_STATE,
		caughtError: false,
	};

	async componentDidMount() {
		// listen to ctr+enter to add new
		document.addEventListener('keydown',this.keydownHandler);
		// load storage state
		const storageState = await storage.load();

		this.setState(
			prevState => ({...prevState, ...storageState})
		);
	}

  componentWillUnmount() {
		// not really needed here ...but just to enforce practice
    document.removeEventListener('keydown', this.keydownHandler);
	}

	componentDidCatch(err, info) {
    this.setState({ caughtError: true });
	}

	// storage api calls
	saveToStore = obj => {
		this.setState(
		 	prevState => ({...prevState, ...obj}),
			() => storage.save(obj)
		);
	};

	// updates noteStyle and/or boardStyle
	updateDefaultStyles = (styleFor='noteStyle', newProp={}) => {
		const updatedProps = {
			...this.state[styleFor],
			...newProp
		};
		this.saveToStore({[styleFor]: updatedProps});
	};

	// reSortNotes = () => {
	// 	const removeFixedStyle = note => ({...note, style: {}});
	// 	const updatedNotes = this.state.notes.map(removeFixedStyle);
	// 	this.saveToStore({notes: updatedNotes});
	// }

	keydownHandler = evt => {
		// ctr + enter
		if ( evt.keyCode === 13 && evt.ctrlKey ) {
			this.addNewNote();
		}
	};

	addNewNote = () => {
		const note = createNote();
		this.saveToStore({notes: [...this.state.notes, note]});
	};

	updateNote = updatedNote => {
		const notes = this.state.notes.map(note => (note.id === updatedNote.id) ? updatedNote : note);
		this.saveToStore({notes});
	};

	removeNote = id  => {
		const notes = this.state.notes.filter(note => note.id !== id);
		this.saveToStore({notes});
	};

	render() {
		if (this.state.caughtError) {
			return <EmojiIcon>Ops! Something went wrong 😥</EmojiIcon>
		}

		const {
			notes,
			noteStyle,
			boardStyle,
		} = this.state;

		const noteBoxes = notes.map( note =>
			<Note
				note={note}
				key={note.id}
				updateFn={this.updateNote}
				removeFn={this.removeNote}
			/>
		);

		return (
			<div
				className='notes-board'
				style={{background: `${boardStyle.daylight ? '' : NIGHT_BG_COLOR}`}}
			>
				<ThemeContext.Provider
					value={{
						defaultNoteStyle: noteStyle,
						defaultBoardStyle: boardStyle,
						updateNoteStyle: newProps => this.updateDefaultStyles('noteStyle' , newProps),
						updateBoardStyle: newProps => this.updateDefaultStyles('boardStyle' , newProps),
					}}
				>
					<ReactCSSTransitionGroup
						transitionName='note'
						transitionAppear
						transitionEnter
						transitionAppearTimeout={500}
						transitionEnterTimeout={500}
						transitionLeaveTimeout={300}
					>
						{noteBoxes}
					</ReactCSSTransitionGroup>

					<EmojiIcon
						onClick={this.addNewNote}
						title='New note (ctr + ⏎)'
						className='plus-sign-emoji'
						aria-label='heavy plus sign'
						data-testid='tid-add-note-btn'
					>
						➕
					</EmojiIcon>

					<SettingsPanel />
				</ThemeContext.Provider>
			</div>
		);
	}
}
