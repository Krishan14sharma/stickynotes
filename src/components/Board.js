import React from 'react';
import extend from 'just-extend';
import Note from './Note';
import './Board.css';


export default class NotesBoard extends React.Component {
	state = {
		notes: [],
	};

	componentDidMount() {
		document.addEventListener('keydown',this.keydownHandler);
		const notes = JSON.parse(localStorage.getItem('ME_NOTES'));
		this.setState({ notes });
	}

  componentWillUnmount() {
    document.removeEventListener('keydown',this.keydownHandler);
	}


	keydownHandler = evt => {
		if ( evt.keyCode === 13 && evt.ctrlKey ) {
			this.addNewNote();
		}
	};

	addNewNote = () => {
		const note = {
			id: Date.now(),
			body: '',
			style: {}
		};

		this.saveNotes([...this.state.notes, note]);
	};

	updateNote = (id, updatedProp) => {
		const notes = this.state.notes.map( note => {
			if ( note.id === id ) {
				extend(note, updatedProp);
				console.info('updated note ', note);
			}
			return note;
		});

		this.saveNotes(notes);
	};

	removeNote = id  => {
		const notes = this.state.notes.filter( note => note.id !== id);

		this.saveNotes(notes);
	};

	saveNotes = notes => {
		this.setState({ notes });
		localStorage.setItem('ME_NOTES', JSON.stringify(notes));
		// chrome.storage.local.set({notes: arr}, function () {

	  //       self.setState({notes: arr});
		// });
		console.log('note : ', this.state.notes);
	};

	render() {
		const { notes } = this.state;

		return (
			<div className='notes-board' >
				{notes.map( note =>
					<Note
						key={note.id}
						note={note}
						updateFn={this.updateNote}
						removeFn={this.removeNote}
					/>
				)}

				<button
					className='add-note-btn'
					onClick={this.addNewNote}
				>
				ADD NOTE
				</button>
			</div>
		);
	}
}
