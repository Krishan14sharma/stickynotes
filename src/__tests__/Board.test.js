import React from 'react';
import { shallow } from 'enzyme';
import Board from '../components/Board';

test('<Board/> renders correctly', () => {
  const component = shallow(<Board />);
  expect(component).toMatchSnapshot();
});
