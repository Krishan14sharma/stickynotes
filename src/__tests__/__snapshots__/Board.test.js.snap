// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotesBoard renders correctly 1`] = `
<div
  className="notes-board"
  style={
    Object {
      "background": "",
    }
  }
>
  <span />
  <span
    aria-label="heavy plus sign"
    className="plus-sign-emoji"
    onClick={[Function]}
    role="img"
    title="New note (ctr + ⏎)"
  >
    ➕
  </span>
  <span
    aria-label="gear sign"
    className="settings-emoji"
    onClick={[Function]}
    role="img"
  >
    ⚙️
  </span>
</div>
`;
